tensorboard/webapp/metrics/store/metrics_store_internal_utils.ts

```diff
@@ -365,6 +380,18 @@ export function generateNextCardStepIndex(
   return nextCardStepIndexMap;
 }
 
+export function generateCardMinMaxStep<T extends PluginType = PluginType>(
+  runsToSeries: RunToSeries<T>
+): MinMaxStep {
+  const allData = Object.values(runsToSeries)
+    .flat()
+    .map((stepDatum) => (stepDatum as StepDatum[T]).step);
```

Comment:
I think you can still get rid of the cast. Just:
` .map((stepDatum) => stepDatum.step); `